### ISSUE TITLE
can't install packages (ex: webpack) with npm starting from 5.4.0

### ISSUE THREAD

**Issue by mgrybyk** [npm/npm#18436]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
npm fails to install package

#### How can the CLI team reproduce the problem?
have npm 5.4.0 or 5.4.1

npm install webpack

NOTE: works fine with npm 5.3.0

**Comment by simonua** [comment]
Not much info here but possible duplicate of #18417?

**Comment by robv8r** [comment]
I've experienced the same problem with **npm 5.4.0 and 5.4.1**.  

**Downgrading to npm 5.3.0 resolves the issue.**

Steps to reproduce:

```
npm install --global npm@5.4.1
npm init -y
npm install webpack
```

It then hangs on the following command for several seconds:

```
> uglifyjs-webpack-plugin@0.4.6 postinstall C:\path\to\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js

[            ......] \ postinstall: info lifecycle webpack@3.5.6~postinstall: webpack@3.5.6
```

After several seconds, the `postinstall` step fails with the following error:

```
npm ERR! path C:\path\to\node_modules\fsevents\node_modules\abbrev\package.json
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall unlink
npm ERR! Error: EPERM: operation not permitted, unlink 'C:\path\to\node_modules\fsevents\node_modules\abbrev\package.json'
npm ERR!     at Error (native)
npm ERR!  { Error: EPERM: operation not permitted, unlink 'C:\path\to\node_modules\fsevents\node_modules\abbrev\package.json'
npm ERR!     at Error (native)
npm ERR!   stack: 'Error: EPERM: operation not permitted, unlink \'C:\\path\\to\\node_modules\\fsevents\\node_modules\\abbrev\\package.json\'\n    at Error (native)',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'unlink',
npm ERR!   path: 'C:\\path\\to\\node_modules\\fsevents\\node_modules\\abbrev\\package.json' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\username\AppData\Roaming\npm-cache\_logs\2017-09-07T20_00_39_708Z-debug.log
```

The file mentioned in the error can be found here: [2017-09-07T20_00_39_708Z-debug.zip](https://github.com/npm/npm/files/1286027/2017-09-07T20_00_39_708Z-debug.zip)

**Comment by robv8r** [comment]
Possible duplicate of #18380

**Comment by robv8r** [comment]
@mgrybyk, What operating system are you on?  I'm on Windows 10 Version 1607.

**Comment by mgrybyk** [comment]
Not sure if it is duplicate, but looks like the the same root cause. I'm using windows 10 1607 with latest update. It's also reproducible on win 2008 server.

Starting from npm 5.4.0 some packages simply can't be installed. Npm 5 is a pain, have to clear caches, delete lock, etc. time to time, but now it's not usable at all.

Thanks

**Comment by rodrigmars** [comment]
Hey @robv8r ... 
!!!Npm 5 is a pain!!! rsrsrs
Your tip solved my problem ... I returned to version 5.3.0 running "npm install --global npm@5.3.0" and now it is possible to install the "webpack". It really does seem to be a problem involving the latest version of npm that affects the installation process by breaking the [\node_modules\fsevents\node_modules\abbrev\package.json ']

Thanks for the tip >>robv8r 
detail ... I'm using windows 7 ...

**Comment by rodrigmars** [comment]
@mgrybyk Thank you for opening this question and sharing the same suffering rsrs

**Comment by lordgreg** [comment]
Exactly the same issue with ionic package. Even after downgrading to npm 5.3.0, I'm now getting the same issue:
```
$ npm i -g ionic@latest
npm WARN Error: EPERM: operation not permitted, lstat 'C:\Users\Gregor\AppData\Roaming\npm\node_modules\ionic\node_modules'
npm WARN     at Error (native)
npm WARN  { Error: EPERM: operation not permitted, lstat 'C:\Users\Gregor\AppData\Roaming\npm\node_modules\ionic\node_modules'
npm WARN     at Error (native)
npm WARN   stack: 'Error: EPERM: operation not permitted, lstat \'C:\\Users\\Gregor\\AppData\\Roaming\\npm\\node_modules\\ionic\\node_modules\'\n    at Error (native)',
npm WARN   errno: -4048,
npm WARN   code: 'EPERM',
npm WARN   syscall: 'lstat',
npm WARN   path: 'C:\\Users\\Gregor\\AppData\\Roaming\\npm\\node_modules\\ionic\\node_modules' }
npm ERR! path C:\Users\Gregor\AppData\Roaming\npm\ionic
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall open
npm ERR! Error: EPERM: operation not permitted, open 'C:\Users\Gregor\AppData\Roaming\npm\ionic'
npm ERR!     at Error (native)
npm ERR!  { Error: EPERM: operation not permitted, open 'C:\Users\Gregor\AppData\Roaming\npm\ionic'
npm ERR!     at Error (native)
npm ERR!   cause:
npm ERR!    { Error: EPERM: operation not permitted, open 'C:\Users\Gregor\AppData\Roaming\npm\ionic'
npm ERR!        at Error (native)
npm ERR!      errno: -4048,
npm ERR!      code: 'EPERM',
npm ERR!      syscall: 'open',
npm ERR!      path: 'C:\\Users\\Gregor\\AppData\\Roaming\\npm\\ionic' },
npm ERR!   stack: 'Error: EPERM: operation not permitted, open \'C:\\Users\\Gregor\\AppData\\Roaming\\npm\\ionic\'\n    at Error (native)',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'open',
npm ERR!   path: 'C:\\Users\\Gregor\\AppData\\Roaming\\npm\\ionic' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.
```

**Comment by mgrybyk** [comment]
@lordgreg, to workaround this:
Stop other npm, node instances

1. Downgrade to 5.3.0
2. Delete node modules folder 
3. Cleanup windows tmp folder

If it doesn't help, try cleaning npm cache and repeat steps 2,3.

If you have some of failing packages installed globally you might need to uninstall them. Make sure they have been fully deleted. Repeat step 3.

**Comment by lordgreg** [comment]
Hi @mgrybyk and thank you for your feedback. Funny, nothing from your list worked. I've found an SO issue, marked as resolved when someone re-installed latest NodeJS. For me, that also fixed the issue. That and only that.

**Comment by mgrybyk** [comment]
@lordgreg, haha. I've tried simply to reinstall node - and it doesn't resolve the issue :)

Probably it is required to cleanup everything and reinstall node...

**Comment by robv8r** [comment]
If you're on Windows, try deleting your node_modules folder.  If that doesn't work, try:

```
npm cache verify
```

If that doesn't work, try running:

```
npm cache clean -force
```

If that doesn't work, uninstall Node.JS, then:

- Delete the folder `%APPDATA%\npm`
  - Contains globally installed packages
  - You may want to make a list of what's in there so you can re-install them later
- Delete the folder `%APPDATA%\npm-cache`
  - Contains cached npm packages.
  - These will be restored as you need them.
- Delete your `node_modules` directory

Now reinstall Node.JS.  Then:

```
npm i -g npm@5.3.0
```

You'll need to re-install any global modules (since you deleted them from `%APPDATA%\npm`.)

**Comment by mgrybyk** [comment]
fixed with npm@5.4.2
related issue:  #18458
commit: 0b28ac72d

**Comment by bluesky66-dev** [comment]
worked after
npm install npm@5.6 -g
